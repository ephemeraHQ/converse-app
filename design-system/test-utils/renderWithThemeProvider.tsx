import { NavigationContainer } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { useThemeProvider } from "@theme/useAppTheme";
import { measureRenders, MeasureRendersOptions } from "reassure";

export const TestThemeProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const {
    themeScheme,
    setThemeContextOverride,
    ThemeProvider,
    navigationTheme,
  } = useThemeProvider();

  return (
    <NavigationContainer theme={navigationTheme}>
      <ThemeProvider value={{ themeScheme, setThemeContextOverride }}>
        {children}
      </ThemeProvider>
    </NavigationContainer>
  );
};

export const renderWithThemeProvider = (ui: React.ReactElement) =>
  render(<TestThemeProvider>{ui}</TestThemeProvider>);

export const measureRendersWithThemeProvider = (
  ui: React.ReactElement,
  options: MeasureRendersOptions
) => measureRenders(<TestThemeProvider>{ui}</TestThemeProvider>, options);
