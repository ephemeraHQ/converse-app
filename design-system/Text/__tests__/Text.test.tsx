// __tests__/Text.test.tsx
import React from "react";
import { Text } from "..";
import type { Text as RNText } from "react-native";
import { renderWithThemeProvider } from "@design-system/test-utils/renderWithThemeProvider";

describe("Text Component", () => {
  it("renders correctly with default props", () => {
    const { toJSON } = renderWithThemeProvider(<Text>Default Text</Text>);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with translation key", () => {
    const { toJSON } = renderWithThemeProvider(
      <Text tx="walletSelector.title" />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("applies weight and size styles", () => {
    const { toJSON } = renderWithThemeProvider(
      <Text weight="bold" size="lg">
        Styled Text
      </Text>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("applies color style", () => {
    const { toJSON } = renderWithThemeProvider(
      <Text color="primary">Colored Text</Text>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("forwards ref correctly", () => {
    const ref = React.createRef<RNText>();
    renderWithThemeProvider(<Text ref={ref}>Ref Text</Text>);
    expect(ref.current).not.toBeNull();
  });

  it("renders children correctly", () => {
    const { getByText } = renderWithThemeProvider(<Text>Child Text</Text>);
    expect(getByText("Child Text")).toBeTruthy();
  });
});
