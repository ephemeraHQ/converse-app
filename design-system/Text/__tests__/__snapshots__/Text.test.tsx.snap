// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component applies color style 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontFamily": "SFProText-Regular",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": 20,
      },
      undefined,
      undefined,
      {
        "color": "#000000",
      },
      undefined,
    ]
  }
>
  Colored Text
</Text>
`;

exports[`Text Component applies weight and size styles 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontFamily": "SFProText-Regular",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": 20,
      },
      {
        "fontFamily": "SFProText-Bold",
        "fontWeight": 700,
      },
      {
        "fontSize": 24,
        "lineHeight": 28,
      },
      undefined,
      undefined,
    ]
  }
>
  Styled Text
</Text>
`;

exports[`Text Component renders correctly with default props 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontFamily": "SFProText-Regular",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": 20,
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Default Text
</Text>
`;

exports[`Text Component renders correctly with translation key 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
        "fontFamily": "SFProText-Regular",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": 20,
      },
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Your messages.
Your privacy.
</Text>
`;
